File: healthmeal/frontend/services/tracking.ts
import { api } from "./api"

export interface FoodItem {
  name: string
  calories: number
  protein: number
  fiber: number
  anti_inflammatory: number
}

export interface FoodLogData {
  id: number
  meal_type: string
  input_method: string
  date: string
  food_items: FoodItem[]
  total_calories: number
  total_protein: number
  total_fiber: number
  anti_inflammatory_score: number
  logged_at: string
}

export interface BodyMetricData {
  id: number
  date: string
  weight: number | null
  body_fat_pct: number | null
  recorded_at: string
}

export interface DailySummary {
  date: string
  total_calories: number
  target_calories: number | null
  total_protein: number
  target_protein: number | null
  total_fiber: number
  anti_inflammatory_score: number
  meal_count: number
  exercise_calories_burned: number
  logs: FoodLogData[]
}

export interface WeeklySummary {
  week_start: string
  week_end: string
  daily_calories: Array<{ date: string; calories: number; exercise: number }>
  avg_protein: number
  avg_fiber: number
  avg_anti_inflammatory: number
  total_exercise_calories: number
}

export interface MonthlySummary {
  month: string
  weekly_calories: Array<{ week: string; avg_calories: number }>
  body_metrics: BodyMetricData[]
  avg_anti_inflammatory: number
  total_days_logged: number
}

export async function addFoodLog(data: {
  meal_type: string
  input_method: string
  date: string
  food_items: FoodItem[]
}): Promise<FoodLogData> {
  const res = await api.post("/food-logs", data)
  return res.data
}

export async function addFoodLogFromPhoto(meal_type: string, image_base64: string): Promise<FoodLogData> {
  const res = await api.post("/food-logs/photo", { meal_type, image_base64 })
  return res.data
}

export async function getFoodLogs(date: string): Promise<FoodLogData[]> {
  const res = await api.get(`/food-logs?date=${date}`)
  return res.data
}

export async function recordBodyMetric(data: {
  date: string
  weight?: number
  body_fat_pct?: number
}): Promise<BodyMetricData> {
  const res = await api.post("/body-metrics", data)
  return res.data
}

export async function getBodyMetrics(): Promise<BodyMetricData[]> {
  const res = await api.get("/body-metrics")
  return res.data
}

export async function getDailySummary(date: string): Promise<DailySummary> {
  const res = await api.get(`/health-summary?period=daily&date=${date}`)
  return res.data
}

export async function getWeeklySummary(date: string): Promise<WeeklySummary> {
  const res = await api.get(`/health-summary?period=weekly&date=${date}`)
  return res.data
}

export async function getMonthlySummary(date: string): Promise<MonthlySummary> {
  const res = await api.get(`/health-summary?period=monthly&date=${date}`)
  return res.data
}

File: healthmeal/backend/schemas/tracking.py
from pydantic import BaseModel
from typing import Optional, List, Any
from datetime import datetime


class NutritionEstimateRequest(BaseModel):
    name: str
    quantity: float
    unit: str


class NutritionEstimateResponse(BaseModel):
    calories: float
    protein: float
    fiber: float
    anti_inflammatory: float


class FoodItemInput(BaseModel):
    name: str
    calories: float
    protein: float
    fiber: float
    anti_inflammatory: float = 5.0


class FoodLogRequest(BaseModel):
    meal_type: str                    # breakfast/lunch/dinner/snack
    input_method: str = "manual"     # manual/photo
    date: str                        # YYYY-MM-DD
    food_items: List[FoodItemInput]


class FoodLogPhotoRequest(BaseModel):
    meal_type: str
    image_base64: str


class FoodLogResponse(BaseModel):
    id: int
    meal_type: str
    input_method: str
    date: str
    food_items: List[dict[str, Any]]
    total_calories: float
    total_protein: float
    total_fiber: float
    anti_inflammatory_score: float
    logged_at: datetime

    class Config:
        from_attributes = True


class BodyMetricRequest(BaseModel):
    date: str
    weight: Optional[float] = None
    body_fat_pct: Optional[float] = None


class BodyMetricResponse(BaseModel):
    id: int
    date: str
    weight: Optional[float]
    body_fat_pct: Optional[float]
    recorded_at: datetime

    class Config:
        from_attributes = True


class DailyHealthSummary(BaseModel):
    date: str
    total_calories: float
    target_calories: Optional[float]
    total_protein: float
    target_protein: Optional[float]
    total_fiber: float
    anti_inflammatory_score: float
    meal_count: int
    exercise_calories_burned: float
    logs: List[FoodLogResponse]


class WeeklyHealthSummary(BaseModel):
    week_start: str
    week_end: str
    daily_calories: List[dict[str, Any]]
    avg_protein: float
    avg_fiber: float
    avg_anti_inflammatory: float
    total_exercise_calories: float


class MonthlyHealthSummary(BaseModel):
    month: str
    weekly_calories: List[dict[str, Any]]
    body_metrics: List[BodyMetricResponse]
    avg_anti_inflammatory: float
    total_days_logged: int
